src/main/java/org/opensearch/security/user/User.java

```diff
@@ -290,10 +290,19 @@ public final Set<String> getSecurityRoles() {
     /**
      * Check the custom attributes associated with this user
      *
-     * @return true if it has a service account attributes. otherwise false
+     * @return true if it has a service account attributes, otherwise false
      */
     public boolean isServiceAccount() {
         Map<String, String> userAttributesMap = this.getCustomAttributesMap();
         return userAttributesMap != null && "true".equals(userAttributesMap.get("attr.internal.service"));
     }
+
+    /**
+     * Check the custom attributes associated with this user
+     *
+     * @return true if it has a plugin account attributes, otherwise false
+     */
+    public boolean isPluginUser() {
+        return name != null && name.startsWith("plugin:");
+    }
```

Comment:
QQ: Do we need to add a restriction that usernames cannot begin with this prefix?

Reply:
I chose this as a convention since `:` is [forbidden](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java#L62-L64) from usernames. Users representing plugins will only exist in-memory while executing transport actions with the plugin's subject. If an inter-nodal transport hop occurs then the user is serialized and then deserialized on the receiving node.